homeassistant/components/heatmiser/climate.py

```diff
@@ -106,9 +107,11 @@ def target_temperature(self):
         """Return the temperature we try to reach."""
         return self._target_temperature
 
-    def set_temperature(self, **kwargs):
+    def set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if ATTR_TEMPERATURE not in kwargs:
+            return
+        temperature = kwargs[ATTR_TEMPERATURE]
```

Comment:
Seems like this fixes a bug too (if SERVICE_SET_TEMPERATURE is called without ATTR_TEMPERATURE).
How about using a walrus here:
```suggestion
        if not (temperature := kwargs.get(ATTR_TEMPERATURE)):
            return
```

Reply:
I think we should only guard against `None`, but the walrus is a good idea.